src/api/firebaseAPI.js

```diff
@@ -0,0 +1,95 @@
+import firebase from 'firebase/app';
+import 'firebase/database'
+
+export class FirebaseAPI {
+  constructor () {
+    this.users = [];
+  }
+
+  createUser (user) {
+    firebase
+      .database()
+      .ref('users/')
+      .push({
+        firstName: user.firstName,
+        lastName: user.lastName,
+        phone: user.phone,
+        email: user.email,
+        id: this.generateId()
+      });
+  }
+
+  generateId () {
+    return Math.floor(10000 + Math.random() * 9000);
+  }
+
+  getUserIndex (id) {
+    return this.users.findIndex(user => user.id === id);
+  }
+
+  updateUser (id, update) {
+    try {
+      const updatedUser = Object.assign(this.getUser(id), update);
+      this.users.splice(this.getUserIndex(id), 1, updatedUser);
+      this.syncData(update);
+    } catch (err) {
+      console.error('Error Editing User: ', err);
+    }
+  }
+
+  syncData (user) {
```

Comment:
I'm not understanding how this is syncing, it looks like it's just deleting?

Reply:
It's adding the user to the db using createUser() and deleting the old record it replaces using deleteUser().